### PR TITLE
chore: add v1 and v2 taraxa capabilities

### PR DESCRIPTION
**Pull request by JakubFornadel**
## Purpose

There is quite a lot of changes in this PR, it implements:

- multiiple taraxa capabilities versions support
- V1 & V2 taraxa capabilities
- optimized votes bundles db storage & network bandwidth
- db migration for period data with optimized votes bundle storage

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,10 +7,10 @@ set(TARAXA_PATCH_VERSION 1)
 set(TARAXA_VERSION ${TARAXA_MAJOR_VERSION}.${TARAXA_MINOR_VERSION}.${TARAXA_PATCH_VERSION})
 
 # Any time a change in the network protocol is introduced this version should be increased
-set(TARAXA_NET_VERSION 1)
+set(TARAXA_NET_VERSION 2)
 # Major version is modified when DAG blocks, pbft blocks and any basic building blocks of our blockchain is modified
 # in the db
-set(TARAXA_DB_MAJOR_VERSION 1)
+set(TARAXA_DB_MAJOR_VERSION 2)
```
Comment (kstdl) [inline CMakeLists.txt:13]: If I am not mistaken this change is minor change, not major. So if we can deal with data format change with migration we should change only minor version and migration will be applied. But on major version change we should rebuild whole db

File: libraries/core_libs/consensus/src/dag/dag_manager.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #include "dag/dag.hpp"
 #include "key_manager/key_manager.hpp"
 #include "network/network.hpp"
-#include "network/tarcap/packets_handlers/dag_block_packet_handler.hpp"
+#include "network/tarcap/packets_handlers/latest/dag_block_packet_handler.hpp"
```
Comment (kstdl) [inline libraries/core_libs/consensus/src/dag/dag_manager.cpp:17]: Do we still need this include here?

**Review by kstdl — APPROVED** [review]

**Review by mfrankovi — APPROVED** [review]

File: libraries/core_libs/network/src/network.cpp [hunk 3]
```diff
@@ -41,35 +63,83 @@ Network::Network(const FullNodeConfig &config, const h256 &genesis_hash,
...
+
+      const size_t kV1NetworkVersion = 1;
+      assert(kV1NetworkVersion < TARAXA_NET_VERSION);
+
+      dev::p2p::Host::CapabilityList capabilities;
+
+      // Register old version (V1) of taraxa capability
+      auto v1_tarcap = std::make_shared<network::tarcap::TaraxaCapability>(
+          kV1NetworkVersion, config, genesis_hash, host, key, packets_tp_, all_packets_stats_, pbft_syncing_state_, db,
+          pbft_mgr, pbft_chain, vote_mgr, dag_mgr, trx_mgr, "V1_TARCAP", network::tarcap::v1::kInitV1Handlers);
```
[the remark this answers is not in the document]
Comment (JakubFornadel) [reply libraries/core_libs/network/src/network.cpp:81]: I am not sure if I understand, but feel free to suggest other naming for the old version ?

[the remark this answers is not in the document]
Comment (JakubFornadel) [reply libraries/core_libs/network/src/network.cpp:81]: true, will change it

**Pull request merged**